Source/Core/Core/PowerPC/JitCommon/JitCache.cpp

```diff
@@ -319,6 +325,23 @@ void JitBaseBlockCache::InvalidateICache(u32 address, const u32 length, bool for
   }
 }
 
+void JitBaseBlockCache::EvictTLBEntry(u32 address)
+{
+  auto iter = start_block_map.lower_bound(address);
+  while (iter != start_block_map.end() && (iter->first & ~0xFFF) == address)
+  {
+    // Remove the block from the fast lookup map, so we re-verify the address
+    // before using it again.
+    iCache[(iter->first >> 2) & iCache_Mask] = 0;
+
+    // Unlink the block.
+    JitBlock& b = blocks[iter->second];
+    WriteDestroyBlock(b.checkedEntry, b.effectiveAddress);
```

Comment:
I don't see why you destroy the block here. IMO it's still valid for the physical address + MSR configuration. We just need to unlink it as the comment says.